web/libs/editor/src/stores/types.d.ts

```diff
@@ -4,7 +4,57 @@ type RawResult = {
   value: object;
 };
 
+type MSTagProps = {
+  isReady?: boolean;
+};
+
+type MSTTagImage = {
+  type: "image";
+  stageWidth: number;
+  stageHeight: number;
+  containerWidth: number;
+  containerHeight: number;
+  canvasSize?: { width: number; height: number };
+} & MSTagProps;
+
+type MSTTag = (
+  | MSTTagImage
+  | {
+      type: string;
+    }
+) &
+  MSTagProps;
+
+type MixinMSTArea = {
+  id: string;
+  ouid: number;
+  results: RawResult[];
+  parentID: string | null;
+};
+
+type MixinMSTRegion = {
+  pid: string;
+  score: number | null;
+  filtered: boolean;
+  parentID: string;
+  fromSuggestion: boolean;
+  dynamic: boolean;
+  origin: "prediction" | "prediction-changed" | "manual";
```

Comment:
I see we have a type enumeration in `web/libs/editor/src/mixins/Regions.js` also - since we need this in 2 spots, would be good to have this type defined and re used in both spots

Reply:
It is quite tricky to try to extract a type from a model and transfer it to a file where imports do not work. I would prefer not to try to solve this problem at the moment.

Reply:
fair enough - would be great to standardize this in the future